test/trento_web/controllers/v1/discovery_controller_test.exs

```diff
@@ -62,5 +62,27 @@ defmodule TrentoWeb.V1.DiscoveryControllerTest do
       assert %DiscardedDiscoveryEvent{payload: ^body, reason: "[:any_error]"} =
                discarded_event
     end
+
+    test "collect action discards application instance registrations when the associated database does not exists",
+         %{conn: conn} do
+      body =
+        load_discovery_event_fixture("sap_system_discovery_application")
+
+      expect(Trento.Commanded.Mock, :dispatch, fn _ ->
+        {:error, :associated_database_not_found}
+      end)
+
+      %{status: status} =
+        conn
+        |> put_req_header("content-type", "application/json")
+        |> post("/api/v1/collect", body)
+
+      assert status == 202
```

Comment:
nitpick: You can pipe this assert in the previous function adding `|> json_response(202)` as well

Reply:
I think leaving this separated makes it clear to the reader that this is an expectation for the test.